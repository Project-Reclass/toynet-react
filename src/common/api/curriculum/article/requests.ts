/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import axios from 'axios';

export interface ArticleImage {
  file: string;
  credit: string;

  caption?: string;
}

export interface ArticleMeta {
  source: string;
  title: string;
  author: string;
  images: ArticleImage[];
}

export const getArticleText = async (articleId: number): Promise<string> => {
  const { data } = await axios.get(`/data/article/${articleId}/article.md`);
  return data;
};

export const getArticleMeta = async (articleId: number): Promise<ArticleMeta> => {
  const { data } = await axios.get(`/data/article/${articleId}/meta.json`);
  return data;
};