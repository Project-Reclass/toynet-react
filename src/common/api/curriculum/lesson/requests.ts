import axios from 'axios';

type LessonSlideURI = string;
export interface Meta {
  numSlides: number;
}

export const getLessonSlide = async (submoduleId: number, slideId: number): Promise<LessonSlideURI> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/${slideId}.png`);
  return data;
};

export const getLessonMeta = async (submoduleId: number): Promise<Meta | undefined> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/meta.json`);
  return data;
};
