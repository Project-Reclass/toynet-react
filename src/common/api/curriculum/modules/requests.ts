import { simulateNetwork } from 'src/common/utils';
import modules from 'src/__data__/curriculum/modules.json';

export const getAllModules = () => {
  return simulateNetwork(() => modules);
};

export const getModuleById = (id: number) => {
  return simulateNetwork(() => modules.find(module => module.id === id));
};

export const getModuleByName = (name: string) => {
  return simulateNetwork(() => modules.find(module => module.name === name));
};