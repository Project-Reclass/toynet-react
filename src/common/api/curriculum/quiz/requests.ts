import axios from 'axios';

export const getQuizMeta = async (quizId: number) => {
	const { data } = await axios.get(`/data/quiz/${quizId}/meta.json`);
	return data;
};