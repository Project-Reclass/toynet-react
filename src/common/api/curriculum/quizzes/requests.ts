import { simulateNetwork } from 'src/common/utils';
import quizzes from 'src/__data__/curriculum/quizzes.json';

export const getAllQuizzes = async () => {
  return simulateNetwork(() => quizzes);
};

export const getQuizById = (id: number) => {
  return simulateNetwork(() => (
    quizzes.find(quiz => quiz.id === id)
  ));
};
