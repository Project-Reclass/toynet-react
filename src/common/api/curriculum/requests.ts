import { simulateNetwork } from 'src/common/utils';
import curriculums from 'src/__data__/curriculum/curriculums.json';

export const getAllCurriculums = () => {
  return simulateNetwork(() => curriculums);
};

export const getCurriculumById = (id: number) => {
  return simulateNetwork(() => (
    curriculums.find(value => value.id === id)
  ));
};