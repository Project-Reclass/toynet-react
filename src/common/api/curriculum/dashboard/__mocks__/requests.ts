/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/

import { DashboardIntf } from 'src/common/types/curriculum';

const defaultMockData = {
  id: 1,
  name: 'test',
  introduction: 'intro',
  modules: [],
};

export let mockData: DashboardIntf = {...defaultMockData};

export function __resetMock() {
  mockData = {...defaultMockData};
}

export function __setMockData(data: DashboardIntf) {
  mockData.id = data.id;
  mockData.name = data.name;
  mockData.introduction = data.introduction;
  mockData.modules = data.modules;
}

export const getCurriculum = async (): Promise<DashboardIntf | undefined> => {
  return Promise.resolve(mockData);
};