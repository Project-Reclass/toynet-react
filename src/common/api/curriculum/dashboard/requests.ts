import axios from 'axios';
import { DashboardIntf } from 'src/common/types/curriculum';

export const getCurriculum = async (curriculumId: number): Promise<DashboardIntf | undefined> => {
  const { data } = await axios.get(`/data/curriculum/${curriculumId}.json`);
  return data;
};