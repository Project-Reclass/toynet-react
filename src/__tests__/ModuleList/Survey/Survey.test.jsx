import React from 'react';
import { renderWithTheme } from 'src/common/test-utils/renderWithTheme';
import { getSurveyMeta } from 'src/common/api/curriculum/survey/requests';
import { MemoryRouter, Route } from 'react-router-dom';
import { waitFor, fireEvent, screen } from '@testing-library/react';
import Survey from 'src/Curriculum/Survey';
jest.mock('src/common/api/curriculum/survey/requests');




const RenderWithRouter = ({ children, moduleId, surveyId }) => (
    <MemoryRouter initialEntries={[`/module/${moduleId}/survey/${surveyId}`]}>
      <Route path="/module/:moduleId/survey/:surveyId">{children}</Route>
    </MemoryRouter>
);

const data = [
  {
      'item_type': 'TEXT',
      'question': 'What is your first name?'
  },
  {
      'item_type': 'CHOICE',
      'question': 'Are you interested in a career in the technology industry?',
      'options': [
          'As of now, I do not plan on it.',
          'Not sure...',
          'I am open to it. :)',
          'Absolutely!'
      ]
  },
  {
      'item_type': 'LONGTEXT',
      'question': 'What do you hope to get out of this course?'
  },
  {
      'item_type': 'SCALE',
      'question': 'How familiar would you say you are with computer networking concepts?',
      'options': [
          'Not at all',
          'Some familiarity',
          'Quite a bit',
          'Very familiar',
          'Professional Experience'
      ]
  }
];

function setInputValue(input, newValue) {
  let lastValue = input.value;
  input.value = newValue;
  let event = new Event("input", { target: input, bubbles: true });
  // React 15
  event.simulated = true;
  // React 16
  let tracker = input._valueTracker;
  if (tracker) {
      tracker.setValue(lastValue);
  }
  input.dispatchEvent(event);
}

describe('The Survey page', () => {
    it('should render the same based on URL parameters', () => {
        const { container } = renderWithTheme(
          <RenderWithRouter moduleId={0} surveyId={6001}>
            <Survey />
          </RenderWithRouter>
        );
        expect(container).toMatchSnapshot();
    });
    it('should fetch quiz data', async () => {
        getSurveyMeta.mockResolvedValue({ items: data });
        const { getByText, getAllByText } = renderWithTheme(
          <RenderWithRouter surveyId={64}>
            <Survey />
          </RenderWithRouter>
        );
    
        await waitFor(() => getAllByText(/first name/i));

        expect(getByText(/first name/i)).toBeInTheDocument();
        expect(getByText(/get out of this course/i)).toBeInTheDocument();
        expect(getByText(/Some familiarity/i)).toBeInTheDocument();
    });
    it('should err if none answered',() => {
      jest.spyOn(window, 'alert').mockImplementation(() => {});
      getSurveyMeta.mockResolvedValue({ items: data });
      const { getByText, getAllByText } = renderWithTheme(
        <RenderWithRouter surveyId={64}>
          <Survey />
        </RenderWithRouter>
      );
      const submitBtn = getByText(/submit/i);
      fireEvent.click(submitBtn);
      expect(window.alert).toBeCalled();
      
    })
    it('should err if not all answered',() => {
      jest.spyOn(window, 'alert').mockImplementation(() => {});
      getSurveyMeta.mockResolvedValue({ items: data });
      const { getByText, getAllByText } = renderWithTheme(
        <RenderWithRouter surveyId={64}>
          <Survey />
        </RenderWithRouter>
      );
      const choice1 = screen.getByTestId('Not sure...');
      const submitBtn = getByText(/submit/i);
      fireEvent.click(choice1);
      fireEvent.click(submitBtn);
      expect(window.alert).toBeCalled();
      
    })
    it('should not err if all answered',() => {
      jest.spyOn(window, 'alert').mockImplementation(() => {});
      getSurveyMeta.mockResolvedValue({ items: data });
      const { getByText, getAllByText } = renderWithTheme(
        <RenderWithRouter surveyId={64}>
          <Survey />
        </RenderWithRouter>
      );
      const choice1 = screen.getByTestId('What is your first name?');
      const choice2 = screen.getByTestId('Not sure...');
      const choice3 = screen.getByTestId('What do you hope to get out of this course?')
      const choice4 = screen.getByTestId('Quite a bit');

      const submitBtn = getByText(/submit/i);

      fireEvent.click(choice2);
      fireEvent.click(choice4);
      setInputValue(choice1,'John Doe');
      setInputValue(choice3,'Nothing');
      fireEvent.click(submitBtn);
      expect(window.alert).not.toBeCalled();
    
    })
    
});