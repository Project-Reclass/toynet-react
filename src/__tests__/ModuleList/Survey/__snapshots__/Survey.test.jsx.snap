// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Survey page should render the same based on URL parameters 1`] = `
<div>
  <div
    class="css-1cxsl7j"
    id="#"
  >
    <div
      class="css-1ikdejk"
    >
      <div
        style="position: relative; height: 100%;"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="css-tcs7sz"
          role="progressbar"
        >
          <svg
            class="css-c0o3jj"
            viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
          >
            <circle
              class="css-9y0bww"
              cx="111.11111111111111"
              cy="111.11111111111111"
              data-progress-track="true"
              r="50"
              stroke-dasharray="314.159"
              stroke-dashoffset="0"
              stroke-width="11.111111111111112"
            />
            <circle
              class="css-146rxdt"
              cx="111.11111111111111"
              cy="111.11111111111111"
              data-progress-indicator="true"
              r="50"
              stroke-dasharray="1 400"
              stroke-dashoffset="0"
              stroke-width="11.111111111111112"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
